src/beast/beast/unit_test/suite.h

```diff
@@ -380,7 +394,9 @@ inline
 void
 suite::operator() (runner& r)
 {
+    *p_this_suite() = this;
     run (r);
+    *p_this_suite() = nullptr;
```

Comment:
Since this is a test suite, I'm assuming `run` can't throw. Even so, if we have a `ScopeExit`-like utility in beast I'd use that here. If not (and `run` can't throw) then fine as-is.